tests/conftest.py

```diff
@@ -610,3 +610,29 @@ def assert_invoke_exited_with(result: Result, exit_code: int):
 
 def assert_invoke_ok(result: Result):
     assert_invoke_exited_with(result, 0)
+
+
+def mocked_requests_unknow_error(*args, **kwargs):
+    class MockResponse:
+        def __init__(self, *args, **kwargs):
+            self.headers = {"content-type": "application/json"}
+            self.status_code = 404
+            self.json_data = {"unknown_detail": "no detail"}
+
+        def json(self):
+            return self.json_data
+
+    return MockResponse()
+
+
+def mocked_requests_scan_error(*args, **kwargs):
+    class MockResponse:
+        def __init__(self, *args, **kwargs):
+            self.headers = {"content-type": "application/json"}
+            self.status_code = 404
+            self.json_data = {"detail": "Not found (404)"}
+
+        def json(self):
+            return self.json_data
```

Comment:
The two `mocked_requests_*` functions are quite similar. Maybe you can factorize these by creating a customizable class to mock requests responses? Something like this:

```python
class MockRequestsResponse:
    def __init__(self, status_code, json_data):
        self.headers = {"content-type": "application/json"}
        self.status_code = status_code
        self.json_data = json_data

        def json(self):
            return self.json_data
```

Then the `test_iac_scan*` functions can be changed like this:

```python
    mocker.patch(
        "ggshield.core.client.IaCGGClient.request",
        return_value=MockRequestsResponse(404, <the_json_dict_here>)
    )
```

Note: I used `return_value` instead of `side_effect` in the call to `mocker.patch()` because we are mocking the return value of the method call here, not a raised exception.

Reply:
Thx for the advices !!